pkg/operator/ceph/cluster/osd/osd.go

```diff
@@ -270,6 +270,64 @@ func (c *Cluster) startProvisioning(config *provisionConfig) {
 			}
 		}
 	}
+
+	// Deploying test based on the storageclassDeviceSet
+	// Todo: Replace the below logic to deploy osd pods over pvcs
+	for setIndex, storageClassDeviceSet := range c.DesiredStorage.StorageClassDeviceSets {
+		for i, pvcTemplate := range storageClassDeviceSet.VolumeClaimTemplates {
```

Comment:
For the first iteration i'd suggest we only support a single volume claim template. This loop looks correct, but for now how about if we break out of the loop after the first one is processed? Specifying multiple templates is just an advanced scenario.